utils/setup.sh

```diff
@@ -21,51 +56,115 @@ MAGENTA_FG="\033[1;35m"
 #CYAN_FG="\033[1;36m"
 RESET="\033[0m"
 
-# Add cloudfront_domain to env file
+#### END variables ####
 
-echo "[*] Updating cloudfront_domain variable in /tools/megazord-composition/.env"
-original_val=$(grep 'CLOUDFRONT_DOMAIN' /tools/megazord-composition/.env)
+#### Utility functions ####
+check_variable() {
+  # Check if a variable is set or not
+  # Parameters:
+  #   $1 - variable to check
+  #   $2 - flag from command-line arguments responsible for this variable
 
-if ! sed -i "s|${original_val}|CLOUDFRONT_DOMAIN=${cloudfront_domain}|" \
-  /tools/megazord-composition/.env; then
-  echo -e "${RED_FG}[ \U2757] Error updating CLOUDFRONT_DOMAIN in /tools/megazord-composition/.env${RESET}"
-  exit 1
-fi
+  if [[ -z "${1}" ]]; then
+    echo -e "${RED_FG}[!]${RESET} ${2} option is required" >&2
+    usage
+    exit 1
+  fi
+}
 
-echo -e "${GREEN_FG}[\U2714] Added CLOUDFRONT_DOMAIN to .env${RESET}\n"
+find_and_replace() {
+  # Search a file for a target value and replaces it with
+  # the updated value
+  # Parameters:
+  #   $1 - target string
+  #   $2 - new value
+  #   $3 - path to file
 
-# Add c2_domain to env file
+  # Get the original line
+  original_val=$(grep "$1" "$3")
 
-echo "[*] Updating C2_DOMAIN variable in /tools/megazord-composition/.env"
-original_val=$(grep 'C2_DOMAIN' /tools/megazord-composition/.env)
+  echo "[*] Updating ${1} variable in ${3}"
 
-if ! sed -i "s|${original_val}|C2_DOMAIN=${domain}|" \
-  /tools/megazord-composition/.env; then
-  echo -e "${RED_FG}[ \U2757] Error updating C2_DOMAIN in /tools/megazord-composition/.env${RESET}"
+  if ! sed -i "s|${original_val}|${1}=${2}|" "${3}"; then
+    echo -e "${RED_FG}[!] Error updating ${1} in ${3}${RESET}"
+    exit 1
+  fi
+
+  echo -e "${GREEN_FG}[\U2714] Added ${1} to ${3}${RESET}\n"
+}
+
+# Get command-line arguments
+while getopts "r:d:c:hp:" arg; do
+  case ${arg} in
+    h)
+      usage
+      exit
+      ;;
+    c)
+      cloudfront_domain=${OPTARG}
+      ;;
+    d)
+      domain=${OPTARG}
+      ;;
+    p)
+      c2_password=${OPTARG}
+      ;;
+    r)
+      redirect_location=${OPTARG}
+      ;;
+    :)
+      echo -e "${RED_FG}[!]${RESET} $0: A value is required for -${OPTARG}" >&2
+      usage
+      exit 1
+      ;;
+    ?)
+      echo -e "${RED_FG}[!]${RESET} $0: Invalid option: -${OPTARG}" >&2
+      usage
+      exit 2
+      ;;
+  esac
+done
+
+if [[ "$#" -eq 0 ]]; then
+  echo -e "${RED_FG}[!]${RESET} Invalid number of arguments" >&2
+  usage
   exit 1
 fi
```

Comment:
Is there a reason this check got moved to after the `getopts` loop?

Reply:
No reason. I guess looking back at it now, it makes more sense to have it before the loop.